demo/src/screens/MainScreen.js

```diff
@@ -50,15 +53,39 @@ class MainScreen extends Component {
     super(props);
 
     const data = props.navigationData || navigationData;
-
     this.state = {
       currentPage: 0,
-      filteredNavigationData: data
+      filteredNavigationData: data,
+      chipsLabels: _.map(data, section => section.title),
+      sectionsData: _.map(data, section => ({title: section.title, data: section.screens})),
+      selectedSection: 0,
+      faderStart: false,
+      faderEnd: true
     };
 
     Navigation.events().bindComponent(this);
   }
 
+  sectionListRef = React.createRef();
+  scrollViewRef = React.createRef();
+
+  onPressFlag = false;
+  onUserScroll = false;
```

Comment:
Let's try improve these flag names 
I wouldn't use `on` prefix cause it hint on an event or a callback name (like `onPress`, `onScroll`)
Try using boolean names like: `hasPressItem`, `hasUserScrolled`